micro-rdk/src/common/entry.rs

```diff
@@ -197,13 +198,17 @@ pub async fn serve_inner<S>(
         network,
         rpc_host,
     )
-    .with_webrtc(webrtc)
-    .with_periodic_app_client_task(Box::new(RestartMonitor::new(restart)))
-    .with_periodic_app_client_task(Box::new(ConfigMonitor::new(
-        *(cfg_response.clone()),
-        storage.clone(),
-        restart,
-    )));
+    .with_webrtc(webrtc);
+
+    let server_builder = server_builder
+        .with_periodic_app_client_task(Box::new(RestartMonitor::new(restart)))
+        .with_periodic_app_client_task(Box::new(ConfigMonitor::new(
+            *(cfg_response.clone()),
+            storage.clone(),
+            restart,
+        )));
+
+    let server_builder = server_builder.with_periodic_app_client_task(Box::new(LogUploadTask {}));
```

Comment:
nit: i get separating the protocol/server stuff from app_client tasks, but unless this is tempoary or would benefit from separated comments, I'd just combine these two.